lib/api.js

```diff
@@ -1984,6 +1986,60 @@ API.prototype.signTxProposalFromAirGapped = function(txp, encryptedPkr, m, n) {
 
 
 /**
+ * Sign transaction proposal from AirGapped
+ *
+ * @param {String} key - A mnemonic phrase or an xprv HD private key
+ * @param {Object} txp
+ * @param {String} unencryptedPkr
+ * @param {Number} m
+ * @param {Number} n
+ * @param {String} network - 'livenet' or 'testnet' (default: 'livenet')
+ * @param {String} passphrase
+ * @param {Number} account - default 0
+ * @param {String} derivationStrategy - default 'BIP44'
+ * @return {Object} txp - Return transaction
+ */
+API.signTxProposalFromAirGapped = function(key, txp, unencryptedPkr, m, n, network, passphrase, account, derivationStrategy) {
+  var self = this;
+
+  var publicKeyRing = JSON.parse(unencryptedPkr);
+
+  if (!_.isArray(publicKeyRing) || publicKeyRing.length != n) {
+    throw new Error('Invalid public key ring');
+  }
+  
+  var newClient = new API({
+    baseUrl: 'https://bws.example.com/bws/api',
+    verbose: false,
+  })
+  
+  if (key.slice(0,4) === 'xprv') {
```

Comment:
Added. Also added sanity check for if user specified 'testnet' but put in an xprv and vice versa.